### PR TITLE
Change business endpoint to filter enrolments

### PR DESCRIPTION
**Pull request by gemmai95**
# Motivation and Context

This change is required to remove logic from action. Action currently has logic to filter enabled and pending enrolments this should live in party.

# What has changed

Allow an optional query param to be passed to the get businesses endpoint. Filter enrolments based on passed status.

# How to test?

Hit endpoint with postman and ensure provided enrolment status.

# Links

Trello: https://trello.com/c/kRD5DTTI

Action PR: https://github.com/ONSdigital/rm-action-service/pull/104

### CODE REVIEW

File: ras_party/controllers/party_controller.py
Comment (shahi645) [inline ras_party/controllers/party_controller.py]: shouldn't this default to `None`?

File: ras_party/controllers/party_controller.py
Comment (shahi645) [inline ras_party/controllers/party_controller.py:112]: a lot of nesting here, maybe consider list comprehension. Although still very readable. Dealers choice i guess

File: ras_party/views/party_view.py
Comment (shahi645) [inline ras_party/views/party_view.py]: untested

File: ras_party/controllers/party_controller.py
Comment (shahi645) [reply ras_party/controllers/party_controller.py]: this is all untested, we should write some unit tests

**Review by shahi645 — CHANGES_REQUESTED** [review]

File: ras_party/controllers/party_controller.py
Comment (ryangrundy7) [reply ras_party/controllers/party_controller.py]: It'll always be passed `None` if an enrolment status isn't there so i don't think theres a point in doing so by here

Comment (shahi645) [reply ras_party/controllers/party_controller.py]: you're right nice

File: ras_party/views/party_view.py
Comment (LJBabbage) [inline ras_party/views/party_view.py]: for consistency this should be enrolment_status

**Review by ryangrundy7 — COMMENTED** [review]
It looks good to me, once tests have been added i'll approve it

File: ras_party/controllers/party_controller.py
Comment (shahi645) [inline ras_party/controllers/party_controller.py]: `id` here is this party id? business id? survey id? Its just named id all the way through from the view to where it needs to use it. Can we be explicit please

**Review by shahi645 — CHANGES_REQUESTED** [review]

File: ras_party/controllers/party_controller.py
Comment (shahi645) [reply ras_party/controllers/party_controller.py]: business id it should be

**Review by andrewmil — APPROVED** [review]
:+1: tested with the action service changes, notifications are now only being generated for ENABLED parties

File: ras_party/controllers/party_controller.py
Comment (gemmai95) [reply ras_party/controllers/party_controller.py]: In this context this can be business party id or respondent party id but I will change it to party id

Comment (shahi645) [reply ras_party/controllers/party_controller.py]: oh really? I couldn't get it to work with a party id

Comment (gemmai95) [reply ras_party/controllers/party_controller.py]: It should work with respondent party id with BI as sample unit type

Comment (shahi645) [reply ras_party/controllers/party_controller.py]: ah i wasn't trying BI as it wasn't in the DB. Ill give that a go

File: test/test_party_controller.py
Comment (shahi645) [inline test/test_party_controller.py:339]: 🌚

File: test/test_party_controller.py
Comment (shahi645) [inline test/test_party_controller.py:348]: why were you getting linting errors for these

Comment (gemmai95) [reply test/test_party_controller.py:348]: Due to the use of the db_session decorator it complains about no value for session

**Review by shahi645 — APPROVED** [review]
LGTM, should probably try and solve the linting errors without the `NOQA` but i dont think it should block this PR

File: test/test_party_controller.py
Comment (shahi645) [reply test/test_party_controller.py:348]: oh this is codacy? yeah its a bitch

Comment (gemmai95) [reply test/test_party_controller.py:348]: Yeah

**Pull request merged**
